mathesar/api/db/viewsets/records.py

```diff
@@ -36,27 +36,28 @@ def list(self, request, table_pk=None):
         table = get_table_or_404(table_pk)
 
         filter_unprocessed = serializer.validated_data['filter']
+
+        column_ids_to_names = table.get_column_name_id_bidirectional_map().inverse
         order_by = serializer.validated_data['order_by']
         grouping = serializer.validated_data['grouping']
-        filter_processed = None
-        column_ids_to_names = table.get_column_name_id_bidirectional_map().inverse
-        if filter_unprocessed:
-            table = get_table_or_404(table_pk)
-            filter_processed = rewrite_db_function_spec_column_ids_to_names(
-                column_ids_to_names=column_ids_to_names,
-                spec=filter_unprocessed,
-            )
         # Replace column id value used in the `field` property with column name
         name_converted_group_by = None
         if grouping:
             group_by_columns_names = [column_ids_to_names[column_id] for column_id in grouping['columns']]
             name_converted_group_by = {**grouping, 'columns': group_by_columns_names}
         name_converted_order_by = [{**column, 'field': column_ids_to_names[column['field']]} for column in order_by]
-
         try:
-
+            filter_processed = _rewrite_db_function_spec(
+                filter_unprocessed, column_ids_to_names,
+            )
```

Comment:
I think this logic would fit much better in the models utility module, `mathesar.utils.models`.